lighthouse-core/gather/connections/connection.js

```diff
@@ -106,11 +106,15 @@ class Connection {
           {method: callback.method, params: object.result}, 'verbose');
         return object.result;
       }));
+    } else if (object.id) {
+      const error = object.error && object.error.message;
+      log.formatProtocol(`disowned method <= browser ${error ? 'ERR' : 'OK'}`,
+          {method: object.method, params: error || object.result}, 'verbose');
```

Comment:
>> normalize object.method here rather than in formatProtocol?

> I'd really rather not, that's exactly what got us into this mess of fatally erroring on unexpected input, unless I'm misunderstanding what you mean by "normalize"

I mean `const method = object.method || '?????';` here instead of in `formatProtocol` *just* for the logging, not for how it's handled.

> where would you like to log the message then?

For better or worse, `formatProtocol` expects a `data.method` and `data.params`, so it's better to be strict in sending that than to just let it support whatever. `formatProtocol`'s existence now is kind of dumb (all it's really doing beyond `log.verbose` is clipping to the end of the line), but rather than eliminating, maybe we can just quickly revise the parameters rather than overload them?

> that would make these fatal in the future which seems bad

yeah, makes sense.